index.js

```diff
@@ -127,39 +128,42 @@ function prettyFactory (options) {
       log = filterLog({ log, ignoreKeys, includeKeys })
     }
 
-    const prettifiedLevel = prettifyLevel({ log, colorizer, levelKey, prettifier: customPrettifiers.level, ...customProps })
-    const prettifiedMetadata = prettifyMetadata({ log, prettifiers: customPrettifiers })
-    const prettifiedTime = prettifyTime({ log, translateFormat: opts.translateTime, timestampKey, prettifier: customPrettifiers.time })
-
     let line = ''
-    if (opts.levelFirst && prettifiedLevel) {
-      line = `${prettifiedLevel}`
-    }
 
-    if (prettifiedTime && line === '') {
-      line = `${prettifiedTime}`
-    } else if (prettifiedTime) {
-      line = `${line} ${prettifiedTime}`
-    }
+    if (!opts.hideMetadata) {
```

Comment:
Can this be done in some other way than continually nesting the core code of this function further and further to the right?

Reply:
Do you have something in mind. I could see some options. One with higher refactoring effort like using strategy pattern (https://refactoring.guru/design-patterns/strategy) and some with lower effort like using switch statements/  lookup tables.

Reply:
I think a simple approach would be to move the now optional code path into a function.

Reply:
Added an approach